dhall-docs/tasty/data/package/JumpToRecordFieldWhenVarIsAnnotatedWithRecordType.dhall

```diff
@@ -0,0 +1,11 @@
+{-|
+On this example, `dhall-docs` will detect that the variable `a` is annotated
+with a record-type, and associate all jump-to-definition features on its field
+access to jump to the fields on that record-type
+
+Note that the `x` field on the record literal isn't highlighted since we consider
+that the variable was _defined_ in the annotation
```

Comment:
It seems like we should prefer the record literal over the record type, when both are present

Reply:
Then the annotation should not affect anything. Is this what you suggest?

Reply:
Yes, that's correct

Reply:
OK, on let-bindings we ignore the type from the annotation on this type-inference

OTOH I believe that in the case of function arguments of record-type from `Lam`, selector expressions should highlight the fields on the record-type. This could be useful to get the associated type of that field in the selector expression when the record-type is huge.

Check https://hydra.dhall-lang.org/build/69827/download/1/docs/JSON/object.dhall.html for example,  hovering the `object` in `json.object` results in the highlight of the `object` field in the record-type, which is really useful IMO

Reply:
@german1608: Yes, I agree that function argument types should be highlighted when possible.  I was only commenting on `let` expressions

Reply:
I'm adding a specific test for `Lam` bindings to reflect that behavior and updating the existing test case for `Let` bindings as well